xbmc/pvr/PVRDatabase.cpp

```diff
@@ -612,7 +594,9 @@ bool CPVRDatabase::Persist(CPVRChannel &channel)
   }
 
   std::string strQuery;
-  if (!ChannelExists(channel.m_iClientId, channel.m_iUniqueId))
+  int existingChannelId = GetChannelId(channel.ClientID(), channel.UniqueID());
+
+  if (existingChannelId == -1)
```

Comment:
in case `GetChannelId` will return a row with an invalid primary id (everything lower 1) we should do the insert.